File: cmd_ban.py
# ملف: cmd_ban.py
import telebot

DEVELOPER_ID = 5860391324

# قائمة الحظر داخل ذاكرة البوت
banned_users = set()

COMMANDS = ["حظر", "عفو", "قائمة_المحظورين"]

def handle(bot, message):
    global banned_users

    uid = message.from_user.id

    # 1️⃣ التحقق من المحظورين
    if uid in banned_users and uid != DEVELOPER_ID:
        bot.reply_to(message, "اُصمت! جاء أمر من الإمبراطور بنفيك خارج مملكته! 👑")
        return

    # 2️⃣ الأوامر الخاصة بالمطور فقط
    if uid != DEVELOPER_ID:
        return

    text = message.text.split()
    cmd = text[0]

    # ===== حظر شخص بالرد =====
    if cmd == "حظر":
        if not message.reply_to_message:
            bot.reply_to(message, "❌ يجب الرد على رسالة الشخص الذي تريد حظره!")
            return
        target_uid = message.reply_to_message.from_user.id
        banned_users.add(target_uid)
        bot.reply_to(message, f"✅ تم حظر المستخدم {message.reply_to_message.from_user.first_name} من مملكة البوت 👑")
        return

    # ===== عفو عن شخص بالرد =====
    if cmd == "عفو":
        if not message.reply_to_message:
            bot.reply_to(message, "❌ يجب الرد على رسالة الشخص الذي تريد العفو عنه!")
            return
        target_uid = message.reply_to_message.from_user.id
        if target_uid in banned_users:
            banned_users.discard(target_uid)
            bot.reply_to(message, f"✅ تم العفو عن المستخدم {message.reply_to_message.from_user.first_name} 👑")
            bot.send_message(target_uid, "بأمر من الإمبراطور تم العفو عنك، اشكره! 👑")
        else:
            bot.reply_to(message, "⚠️ هذا الشخص ليس محظوراً بالفعل!")
        return

    # ===== عرض قائمة المحظورين =====
    if cmd == "قائمة_المحظورين":
        if not banned_users:
            bot.reply_to(message, "📜 لا يوجد مستخدمين محظورين حالياً.")
            return
        text_list = "📜 قائمة المحظورين الحاليين:\n"
        for u in banned_users:
            text_list += f"• {u}\n"
        bot.reply_to(message, text_list)
        return

File: cmd_ban.js
// cmd_ban.js
const db = require('./db_manager');

const DEV_IDS = [5860391324, 7076215547, 7855813063]

const COMMANDS_BAN = ["حظر"];
const COMMANDS_PARDON = ["عفو"];
const COMMANDS_LIST = ["قائمة الحظر", "الحظر"];

function handle(bot, ctx) {
    const uid = ctx.from.id;
    const user_name = ctx.from.first_name;
    const text = ctx.message?.text?.trim();

    if (!text) return;

    // ===== حظر المستخدم =====
    if (COMMANDS_BAN.some(cmd => text.startsWith(cmd))) {
        if (uid !== DEV_ID) return;
        if (!ctx.message.reply_to_message) {
            ctx.reply("⚠️ الرجاء الرد على رسالة الشخص الذي تريد حظره!");
            return;
        }
        const target_id = ctx.message.reply_to_message.from.id;
        db.ban_user(target_id);
        ctx.reply(`👑 ${user_name} أمر الإمبراطور: تم حظر الشخص بنجاح!`);
        return;
    }

    // ===== العفو عن المستخدم =====
    if (COMMANDS_PARDON.some(cmd => text.startsWith(cmd))) {
        if (uid !== DEV_ID) return;
        if (!ctx.message.reply_to_message) {
            ctx.reply("⚠️ الرجاء الرد على رسالة الشخص الذي تريد العفو عنه!");
            return;
        }
        const target_id = ctx.message.reply_to_message.from.id;
        db.unban_user(target_id);
        ctx.reply(`👑 ${user_name} أمر الإمبراطور: تم العفو عن الشخص!`);

        // إرسال رسالة للعفو عنه
        try {
            bot.telegram.sendMessage(target_id, "👑 تم العفو عنك بأمر الإمبراطور! اشكر الإمبراطور 🌟");
        } catch {}
        return;
    }

    // ===== عرض قائمة الحظر =====
    if (COMMANDS_LIST.includes(text)) {
        if (uid !== DEV_ID) {
            ctx.reply("❌ هذا الأمر للمطور فقط!");
            return;
        }
        const banned_users = db.get_banned_users ? db.get_banned_users() : [];
        if (!banned_users.length) {
            ctx.reply("📜 لا يوجد أي شخص محظور حالياً.");
            return;
        }
        let text_list = "📜 قائمة المحظورين:\n\n";
        banned_users.forEach(u => text_list += `👤 ID: ${u}\n`);
        ctx.reply(text_list);
        return;
    }

    // ===== الصمت لأي شخص محظور =====
    if (db.is_user_banned(uid)) {
        try {
            ctx.deleteMessage(ctx.message.message_id).catch(() => {});
        } catch {}
        return;
    }
}

module.exports = { handle };
